src/Data/Str.php

```diff
@@ -169,4 +169,31 @@ public static function ConvertCase(string $str, $encoding = null)
         // returns original string when mbstring is not active.
         return $str;
     }
+
+    /**
+     * Checks if given string is a valid Base64 encoded string
+     *
+     * @param string $str String to be tested
+     *
+     * @return bool wether it is a valid base64 encoded string or not
+     */
+    public static function isBase64(string $str): bool
+    {
+        return ($str === base64_encode(base64_decode($str)));
+    }
+
+    /**
+     * Return only a portion of given string
+     *
+     * @param string $str input string to process
+     * @param int $start where to start the cut
+     * @param int|null $length how many characters to return
+     * @param string $encoding optional encoding to use
+     *
+     * @return string
+     */
+    public static function substring(string $str, int $start, int $length = null, string $encoding = 'UTF-8'): string
+    {
+        return mb_substr($str, $start, $length, $encoding);
```

Comment:
the last argument should be like `self::encoding($encoding);`

The value of encoding argument in function should be null or `?$encoding`

Reply:
thanks for the hint, totally overlooked how it was done in the other functions... 

should be fixed now